### ISSUE TITLE
Consider using C instead of ObjC for macOS/iOS

### ISSUE THREAD

**Issue by paulherman** [floooh/sokol#754]
TL;DR: one could use plain C with lots of casts to write the same code in C as in ObjC.

Some examples and references:
- https://github.com/halide/Halide/blob/main/src/runtime/objc_support.h
- https://www.mikeash.com/pyblog/objc_msgsends-new-prototype.html

Of course, this comes at quite a high readability cost. The motivation I have behind this idea is a simplification at build time:
- in Go, the foreign function interface requires `-x objective-c` as an extra flag
- some build systems (e.g. Bazel) require different rules depending on the language (cc_library vs objc_library per https://bazel.build/reference/be/objective-c)

For completeness, I'm mostly wondering if you think this is worth doing in the first place, rather than asking you to do it. Perhaps an issue is not the best way to discuss this :)

**Comment by floooh** [comment]
I've been considering this and even started tinkering in that direction (https://github.com/floooh/objc-ast-experiments), but it's not really realistic unless code generation like in the metal-cpp headers is used (but this doesn't cover any Cocoa APIs).

There are also subtle ABI differences between x86-64 and ARM64 which need to be considered in such a C API wrapper, but which I haven't investigated yet (it seems the ARM64 ObjC runtime is missing one of the ```objc_msgSend_stret``` or ```objc_msgSend_fpret``` functions).

IME, build systems usually 'do the right thing' if you place the sokol implementation into a .m file (for instance in the sokol samples I'm using different implementation source files depending on platform: https://github.com/floooh/sokol-samples/tree/master/libs/sokol)
